api/internal/conf/conf.go

```diff
@@ -249,3 +254,16 @@ func initEtcdConfig(conf Etcd) {
 		Prefix:    prefix,
 	}
 }
+
+// initialize parallelism settings
+func initParallelism(choiceCores int) {
+	if choiceCores < 0 {
+		return
+	}
+	maxSupportedCores := runtime.NumCPU()
+
+	runtime.GOMAXPROCS(maxSupportedCores)
+	if choiceCores < maxSupportedCores {
+		runtime.GOMAXPROCS(choiceCores)
+	}
```

Comment:
> GOMAXPROCS sets the maximum number of CPUs that can be executing simultaneously and returns the previous setting. It defaults to the value of runtime.NumCPU. If n < 1, it does not change the current setting. This call will go away when the scheduler improves.

As the doc said, I think this will be more clear.

```golang
maxSupportedCores := runtime.NumCPU()
if choiceCores > maxSupportedCores {
      choiceCores = maxSupportedCores
}

runtime.GOMAXPROCS(choiceCores)
```

Reply:
yeah, sure. Thanks for the review.